src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Statistics.cs

```diff
@@ -407,15 +407,20 @@ internal static IPInterfaceStatisticsTable ParseInterfaceStatisticsTableFromFile
             {
                 sr.ReadLine();
                 sr.ReadLine();
-                int index = 0;
+                Span<Range> pieces = stackalloc Range[18]; // [0]-[16] used, +1 to ensure any additional segment goes into [17]
                 while (!sr.EndOfStream)
                 {
                     string line = sr.ReadLine()!;
                     if (line.Contains(name))
                     {
-                        Span<Range> pieces = stackalloc Range[18]; // [0] skipped, [1]-[16] used, +1 to ensure any additional segment goes into [17]
                         ReadOnlySpan<char> lineSpan = line;
-                        pieces = pieces.Slice(0, lineSpan.SplitAny(pieces, " :", StringSplitOptions.RemoveEmptyEntries));
+                        lineSpan.SplitAny(pieces, " :", StringSplitOptions.RemoveEmptyEntries);
```

Comment:
```suggestion
                        int pieceCount = lineSpan.SplitAny(pieces, " :", StringSplitOptions.RemoveEmptyEntries);

                        if (pieceCount < 17)
                        {
                            continue;
                        }
```

Since we're ignoring the count returned here, we could end up reading a range from a previous iteration if we encounter invalid input, whereas before we would have thrown an `IndexOutOfRangeException`.